web/packages/teleterm/src/ui/Search/SearchContext.tsx

```diff
@@ -76,16 +80,21 @@ export const SearchContextProvider: FC = props => {
   const close = useCallback(() => {
     setIsOpen(false);
     setActivePicker(actionPicker);
-    if (previouslyActive.current instanceof HTMLElement) {
+    if (previouslyActive.current) {
       previouslyActive.current.focus();
     }
   }, []);
 
+  const closeWithoutRestoringFocus = useCallback(() => {
+    previouslyActive.current = undefined;
+    close();
+  }, [close]);
```

Comment:
I wish I didn't have to implement it like this, I'd rather have `close` which doesn't restore focus and then another version of `close` which does. This way we wouldn't have to depend on some state interactions to achieve the effect we want.

But given that in all but one case we want to restore focus, I just couldn't find a way to do this without making the code unnecessarily complicated.